executor/union_scan.go

```diff
@@ -117,8 +117,27 @@ func (us *UnionScanExec) Open(ctx context.Context) error {
 	if err := us.baseExecutor.Open(ctx); err != nil {
 		return err
 	}
+	return us.open(ctx)
+}
+
+func (us *UnionScanExec) open(ctx context.Context) error {
+	var err error
+	reader := us.children[0]
+	switch x := reader.(type) {
+	case *TableReaderExecutor:
+		us.addedRows, err = buildMemTableReader(us, x).getMemRows()
```

Comment:
why not return the err here?

Reply:
Open `UnionScanExec` need do `us.snapshotChunkBuffer = us.newFirstChunk()` in line #139.

Reply:
If there is an error here, we also need to init the chunk?

Reply:
No, I add err check in line #138 now.